File: DynamicPrgramming/343. Integer Break/Solution.py
class Solution(object):
    def integerBreak(self, n):
        """
        :type n: int
        :rtype: int
        """
        f = [0 for i in range(n+1)]
        f[1]=1
        for i in range(2, n+1):
            if i<n:
                f[i]=i
            for j in range(1, i/2+1):
                f[i]=max(f[i], f[j]*f[i-j])
        return f[n]

File: DynamicPrgramming/343. Integer Break/Solution.java
class Solution {
    public int integerBreak(int n) {
        int[] f = new int[n+1];
        f[0]=Integer.MIN_VALUE; //impossible
        f[1]=1;
        for (int i=2; i<=n; i++){
            f[i]=(i==n)?1:i;
            for (int j=1; j<=i/2; j++){
                f[i]=Math.max(f[i], f[j]*f[i-j]);
            }
        }
        return f[n];
    }
}